app/bundles/EmailBundle/Form/Type/EmailSendType.php

```diff
@@ -186,6 +195,30 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 );
             }
         }
+        if (!empty($options['with_immediately'])) {
+            if ($this->coreParametersHelper->getParameter('mailer_spool_type') === 'file') {
+                $default = (isset($options['data']['immediately'])) ? $options['data']['immediately'] : false;
+                $builder->add(
+                'immediately',
+                YesNoButtonGroupType::class,
+                [
+                    'label' => 'mautic.form.action.sendemail.immediately',
+                    'data'  => $default,
+                    'attr'  => [
+                        'tooltip'=> 'mautic.form.action.sendemail.immediately.desc',
+                    ],
+                ]
+            );
+            } else {
```

Comment:
About that comment about testing the `else` part, I meant that you have to write very similar second test so one test would test the `if` part and the other the `else` part. The mock of `$this->coreParametersHelper->getParameter('mailer_spool_type')` will return `file` in one test and something else in the second test.

Reply:
Ok, I created testSetSampleMailerByTemporaryMailerSpoolFile  and testSetSampleMailerByTemporaryMailerSpoolMemory
Also I created setUp with mockups, but not time for refactor all class , just my method :(

Reply:
That's OK. Focus just on your changes. But it shouldn't work the way you did it. That means that the changed code is not tested if you can test both options with one test. I'm looking closely on the test. You should call the `buildForm` method with `FormBuilderInterface` mock as the first argument. Then check what will be added to the builder. And that's the important part. Because if the `mailer_spool_type` === `file` then it will add different params than it it won't be `file`.

Reply:
Please help, I've added it but can't move forward

https://github.com/mautic/mautic/pull/5566/files#diff-27b27bab078613ba4d7a049eccb75123R830 
https://github.com/mautic/mautic/pull/5566/files#diff-27b27bab078613ba4d7a049eccb75123R841

Reply:
Hello @kuzmany,

There is some misunderstanding I think:
1) We are talking about `EmailSendType` class here, but you linked to test of `SendEmailToContact` class.
2) Just add 2 new tests for this class - first return FALSE and second return TRUE. You will assert that correct call on a builder was performed.

Reply:
@Maxell92 
 I don't understand. EmailSendType class - there is not TRUE/FALSE logic?  
I am lost now, what test do you need add to PR?

Reply:
Hello @kuzmany,
You want to test if / else condition - to be sure correct calls on builder were performed. So there should be 2 tests for the `if (!empty($options['with_immediately'])) {` condition - one for TRUE, second for FALSE.

Reply:
@Maxell92 I am building the test

https://github.com/kuzmany/mautic/blob/da8d1634cde20749a9a3d2cbf13916da0e0e5e55/app/bundles/EmailBundle/Tests/Form/Type/EmailSendTypeTest.php 

Can you help how can I provide check If formBuilder generate YesNoButtonGroupType or HiddenType field? 

Thanks

Reply:
I'm not sure I'm following this, but is this something we can screenshare and tag-team to get this issue merged?

Reply:
@kuzmany I think he's suggesting you add one more test after testSendEmailFormBasedOnSpoolType without with_immediately set.